server-spi-private/src/main/java/org/keycloak/models/policy/ResourcePolicyProvider.java

```diff
@@ -42,4 +37,86 @@ public interface ResourcePolicyProvider extends Provider {
      * This method checks a list of candidates and returns only those that are eligible based on time.
      */ 
     List<String> filterEligibleResources(List<String> candidateResourceIds, long time);
+
+    /**
+     * Indicates whether the specified resource is in the scope of this policy. For example, a policy associated with a
+     * broker is applicable only to users with a federated identity associated with the same broker.
+     *
+     * @param resourceId the id of the resource being checked.
+     * @return {@code true} if the resource is in the policy scope; {@code false} otherwise.
+     */
+    boolean isResourceInScope(String resourceId);
```

Comment:
As discussed, perhaps we can simplify the contract and rely on the `supportsSchedulingOnEvent` to also check whether a resource is in the scope (origin/source) of a policy.

The ` ResourcePolicyEvent` already provides the resource id.